File: backend/Apex/src/Apex.Invest.Features/Modules/Bonds/Details/Models/BondDetailsModel.cs
using Apex.Invest.Domain.Enums;

namespace Apex.Invest.Features.Modules.Bonds.Details.Models;

public class BondDetailsModel
{
    public int Id { get; set; }
    public int PlatformId { get; set; }
    public string Ticker { get; set; } = default!;
    public string Issuer { get; set; } = default!;

    public Currency Currency { get; set; }
    public decimal ParPrice { get; set; }
    public decimal CouponRate { get; set; }
    public int NextCouponDate { get; set; }
    public DateTime MaturityDate { get; set; }
    public PaymentFrequence PaymentFrequence { get; set; }
    public Status Status { get; set; }

    public int TotalQuantity { get; set; }
    public decimal AveragePurchasePrice { get; set; }
    public decimal TotalInvested { get; set; }

    public List<BondOperationModel>? Operations { get; set; }
}

File: frontend/apex-invest/src/app/features/bond/pages/bond-details-page/models/bond-details.model.ts
import { BondDetailsOperation } from "./bond-details-operations";

export interface BondDetails {
  id: number;
  platformId: number;
  ticker: string;
  issuer: string;
  currency: string;
  parPrice: number;
  couponRate: number;
  paymentFrequency: string;
  nextCouponDate: number;
  maturityDate: string;
  status: string;
  operations: BondDetailsOperation[];
}